controllers/programmes/index.js

```diff
@@ -208,51 +208,75 @@ router.get(
     '/:programmeSlug/:childPageSlug?',
     injectFundingProgramme,
     (req, res, next) => {
-        const { fundingProgramme } = res.locals;
-
-        if (get(fundingProgramme, 'entryType') === 'contentPage') {
-            setCommonLocals({ res, entry: fundingProgramme });
-            set(
-                res.locals,
-                'content.flexibleContent',
-                fundingProgramme.content
-            );
-            const parentProgrammeCrumb = fundingProgramme.parent
-                ? {
-                      label: fundingProgramme.parent.title,
-                      url: fundingProgramme.parent.linkUrl
-                  }
-                : {};
-            res.render(
-                path.resolve(__dirname, '../common/views/flexible-content'),
-                {
-                    breadcrumbs: concat(res.locals.breadcrumbs, [
+        const { currentPath, fundingProgramme } = res.locals;
+
+        /* Only render if not using an external path */
+        if (
+            fundingProgramme &&
+            startsWith(currentPath, fundingProgramme.linkUrl)
+        ) {
+            if (get(fundingProgramme, 'entryType') === 'contentPage') {
```

Comment:
All of the logic inside this statement is duplicates in the Strategics controller – I tried to extract it somewhere shared but couldn't work out where to put it... `/common`?